js/demo-app/public/index.html

```diff
@@ -14,11 +14,55 @@
     </noscript>
     <div class="root-container">
         <div class="nav">
+            <span>With URL routing:</span>
+            <br/>
             <a class="nav-item" href="#/one">Component 1</a>
             <a class="nav-item" href="#/two">Component 2</a>
+            <button id="toggle" onclick="toggleRouting()" style="float: right;">Enable Routing</button>
+        </div>
+        <div class="nav">
+            <span>Programmatically:</span>
+            <br/>
+            <button class="nav-item" onclick="setRoute('#/one')">Component 1</button>
+            <button class="nav-item" onclick="setRoute('#/two')">Component 2</button>
         </div>
         <frame-router id="router"></frame-router>
     </div>
     <script type="text/javascript" src="/app.js"></script>
+    <script>
```

Comment:
This is a nit, but you could move all this js into the app.js file.  This will make development a bit easier in editors.  

You could also use the event listener on `DOMContentLoaded` to find the toggle, router, and nav items and attache the listeners.  This will also avoid polluting the global namespace.

Reply:
I moved this part to index.js file.
I'm merging this, and we will enhance the demo-app next time :)